jetty-io/src/main/java/org/eclipse/jetty/io/MappedByteBufferPool.java

```diff
@@ -93,37 +123,41 @@ public void release(ByteBuffer buffer)
         if (buffer == null)
             return; // nothing to do
 
-        // validate that this buffer is from this pool
-        assert ((buffer.capacity() % _factor) == 0);
+        int capacity = buffer.capacity();
+        // Validate that this buffer is from this pool.
+        assert ((capacity % getCapacityFactor()) == 0);
 
-        int b = bucketFor(buffer.capacity());
+        int b = bucketFor(capacity);
         ConcurrentMap<Integer, Bucket> buckets = bucketsFor(buffer.isDirect());
+        Bucket bucket = buckets.computeIfAbsent(b, _newBucket);
 
-        Bucket bucket = buckets.computeIfAbsent(b, this::newBucket);
-        bucket.release(buffer);
+        if (incrementMemory(buffer))
+            bucket.release(buffer);
```

Comment:
again we need to either clear entire pool or at least make a lame attempt at finding LRU buffer